jepsen/scalardb/src/scalardb/transfer_append.clj

```diff
@@ -0,0 +1,322 @@
+      :get-num-records (if-let [result (let [tx (scalar/start-transaction test)]
+                                         (->> (range (:num op))
+                                              (map prepare-scan)
+                                              (map #(.scan tx %))
+                                              (map count)
+                                              (reduce +)))]
+                          (assoc op :type :ok, :value result)
+                          (assoc op :type :fail, :error "Failed to get number of records"))
+      :check-tx (let [unknown (:unknown-tx test)]
+                  (if-let [num-committed (if (empty? @unknown)
+                                           0
+                                           (scalar/check-transaction-states test @unknown))]
+                    (assoc op :type :ok, :value num-committed)
+                    (assoc op :type :fail, :error "Failed to check status")))))
+
+  (close! [_ _])
+
+  (teardown! [_ test]
+    (scalar/close-all! test)))
+
+(defn transfer
+  [test _]
+  (let [n (-> test :model :num)]
+    {:type :invoke
+     :f :transfer
+     :value {:from (rand-int n)
+             :to (rand-int n)
+             :amount (+ 1 (rand-int 1000))}}))
+
+(def diff-transfer
+  (gen/filter (fn [op] (not= (-> op :value :from)
+                             (-> op :value :to)))
+              transfer))
+
+(defn get-all
+  [test _]
+  {:type :invoke
+   :f :get-all
+   :num (-> test :model :num)})
+
+(defn get-num-records
+  [test _]
+  {:type :invoke
+   :f :get-num-records
+   :num (-> test :model :num)})
+
+(defn check-tx
+  [test _]
+  {:type :invoke
+   :f :check-tx})
+
+(defn consistency-checker
+  []
+  (reify checker/Checker
+    (check [this test history opts]
+      (let [read-result (->> history
+                             (r/filter op/ok?)
+                             (r/filter #(= :get-all (:f %)))
+                             (into [])
+                             last
+                             :value)
+            actual-balance (->> read-result
+                                (map :balance)
+                                (reduce +))
+            bad-balance (if-not (= actual-balance total-balance)
+                          {:type     :wrong-balance
+                           :expected total-balance
+                           :actual   actual-balance})
+            actual-age (->> read-result
+                            (map :age)
+                            (reduce +))
+            expected-age (let [num-records (->> history
+                                                (r/filter #(= :get-num-records (:f %)))
+                                                (into [])
+                                                last
+                                                :value)]
+                           (- num-records (:num (:model test)))) ; age starts at 0 so should be num-records minus num-accounts
```

Comment:
If you change the first value of each age to `1` in `populate-accounts`, this compensation isn't needed?

Reply:
Good idea. This makes it a bit more clean. Thank you.